packages/plugin-ext/src/main/browser/plugin-contribution-handler.ts

```diff
@@ -290,4 +291,43 @@ export class PluginContributionHandler {
         }
         return result;
     }
+
+    protected validateConfigurationSchema(schema: PreferenceSchema): void {
+        // tslint:disable-next-line:forin
+        for (const p in schema.properties) {
+            const property = schema.properties[p];
+            if (property.type !== 'object') {
+                continue;
+            }
+
+            if (!property.default) {
+                this.validateDefaultValue(property);
+            }
+
+            const properties = property['properties'];
+            if (properties) {
+                // tslint:disable-next-line:forin
+                for (const key in properties) {
+                    if (typeof properties[key] !== 'object') {
+                        delete properties[key];
+                    }
+                }
+            }
+        }
+    }
+
+    private validateDefaultValue(property: PreferenceSchemaProperties): void {
+        property.default = {};
```

Comment:
@tolusha that's not correct, `null` is valid value, if we change it then some VS Code extension won't work properly, since they will get an object instead of `null`. Do you remember why we did it? I think we should not mutate schemas from the manifest file at all.

Reply:
I don't remember actually.